lib/wasmer-types/src/values.rs

```diff
@@ -204,3 +222,78 @@ impl<T> From<ExternRef> for Value<T> {
 //         Self::FuncRef(val)
 //     }
 // }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_value_i32_from_u32() {
+        let input: u32 = 0x00000000;
+        let value = Value::<()>::from(input);
+        let mut out: i128 = 0;
+        unsafe {
+            value.write_value_to(&mut out as *mut i128);
+        }
+        assert_eq!(out, 0x00000000);
+
+        let input: u32 = 0x00000001;
+        let value = Value::<()>::from(input);
+        let mut out: i128 = 0;
+        unsafe {
+            value.write_value_to(&mut out as *mut i128);
+        }
+        assert_eq!(out, 0x00000001);
+
+        let input: u32 = 0xaabbccdd;
+        let value = Value::<()>::from(input);
+        let mut out: i128 = 0;
+        unsafe {
+            value.write_value_to(&mut out as *mut i128);
+        }
+        assert_eq!(out, 0xaabbccdd);
+
+        let input: u32 = 0xffffffff;
+        let value = Value::<()>::from(input);
+        let mut out: i128 = 0;
+        unsafe {
+            value.write_value_to(&mut out as *mut i128);
+        }
+        assert_eq!(out, 0xffffffff);
+    }
+
+    #[test]
+    fn test_value_i64_from_u64() {
+        let input: u64 = 0x0000000000000000;
+        let value = Value::<()>::from(input);
+        let mut out: i128 = 0;
+        unsafe {
+            value.write_value_to(&mut out as *mut i128);
+        }
+        assert_eq!(out, 0x0000000000000000);
+
+        let input: u64 = 0x0000000000000001;
+        let value = Value::<()>::from(input);
+        let mut out: i128 = 0;
+        unsafe {
+            value.write_value_to(&mut out as *mut i128);
+        }
+        assert_eq!(out, 0x0000000000000001);
+
+        let input: u64 = 0xaabbccddeeff0011;
+        let value = Value::<()>::from(input);
+        let mut out: i128 = 0;
+        unsafe {
+            value.write_value_to(&mut out as *mut i128);
+        }
+        assert_eq!(out, 0xaabbccddeeff0011);
+
+        let input: u64 = 0xffffffffffffffff;
```

Comment:
Right, the idea is to consistently use a byte notation for all tests. Also I need the same value for the input and expeted output. Can I put `u64::max_value()` in an u128? Maybe that works.

I can change that, but would favour the same test layout everywhere.

Reply:
I think we can probably avoid any use of `u128` in these tests.  We can test the low level part separately but I think if the conversion from u32 -> i32 works in the context of Value then we can test that Value converts to the correct bytes in a u128 separately.

If there was more `unsafe` going on, then I think it would make sense to test these together, but as is, I just see this as a way to make sure that we don't break conversion logic.